### PR TITLE
refactor: Port ScoreboardHandler to StyledText

### PR DESCRIPTION
**Pull request by Mcrtin**

### CODE REVIEW

File: common/src/main/java/com/wynntils/models/spells/actionbar/SpellSegment.java [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
 public class SpellSegment implements ActionBarSegment {
     private static final Pattern SPELL_PATTERN =
-            Pattern.compile("§0 +§a([RL])§7-(?:§[a7n])+([RL?])§7-§r(?:§[a7n])+([LR?])§r +");
+            Pattern.compile("§0 +§a([RL])§7-(?:§[a7n])+([RL?])§7-(?:§[a7n])+([LR?]) +");
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/models/spells/actionbar/SpellSegment.java]: How is this relevant to scoreboards?

**Review by kristofbolyai — COMMENTED** [review]
Looks good. Did you check all scoreboard part files for patterns (iirc they only have simple formatting though, so it shouldn't be a problem)?

**Comment by Mcrtin** [comment]
I didn't check the ScoreboardPart inheritors but now i did and there isn't anything to do with the patterns.

**Review by kristofbolyai — APPROVED** [review]
Looks good!
